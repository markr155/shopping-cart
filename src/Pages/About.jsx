export default function About() {
	return (
		<>
			<p>
				This fake store was created as a project for The Odin Project
				curriculum.
			</p>
			<p>
				The main skills learned while implementing this project are prop type validation, React
				Router, including Routed paths, Outlet and OutletContext, and CSS using
				CSS-in-JS via the styled-components library.
				
			</p>
			<p>
				Creating this project was done via Test Driven Development (TDD) and the
				React Testing Library.
			</p>
		</>
	);
}
