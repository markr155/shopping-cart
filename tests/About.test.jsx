import About from '../src/Pages/About';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('About renders', () => {
	const { container } = render(<About />, {
		wrapper: BrowserRouter,
	});
	it('renders elements', () => {
		expect(container).toMatchSnapshot();
	});
});
