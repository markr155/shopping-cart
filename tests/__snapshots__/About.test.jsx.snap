// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About renders > renders elements 1`] = `
<div>
  <h1>
    About us page
  </h1>
  <p>
    This fake store was created as a project for The Odin Project curriculum.
  </p>
  <p>
    The main skills learned while implementing this project are using React Router, new CSS using CSS-in-JS via the styled-components library.
  </p>
</div>
`;
